include/revng/Pipes/RevngPasses.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/Pass.h"
+
+#include "revng/Model/Function.h"
+
+namespace pipeline {
+class ExecutionContext;
+class TargetsList;
+
+class RevngModulePass : public llvm::ModulePass {
+  using llvm::ModulePass::ModulePass;
+
+  bool runOnModule(llvm::Module &Module) final;
+
+  virtual bool runOnTargets(llvm::Module &Module,
+                            const TargetsList *Targets) = 0;
+};
+
+class RevngFunctionPipe {
```

Comment:
1. Rename to `FunctionPassImpl`
2. Rename `RevngFunctionPass` to `FunctionPass`